license-maven-plugin/src/main/java/com/mycila/maven/plugin/license/header/HeaderDefinition.java

```diff
@@ -61,13 +64,40 @@ public HeaderDefinition(String type,
                           String skipLinePattern,
                           String firstLineDetectionPattern, String lastLineDetectionPattern,
                           boolean allowBlankLines, boolean isMultiline, boolean padLines) {
+     this(type, firstLine, beforeEachLine, endLine, afterEachLine, skipLinePattern, firstLineDetectionPattern, null, lastLineDetectionPattern, allowBlankLines, isMultiline, padLines);
+  }  
+  
+  /**
+   * Constructs a new <code>HeaderDefinition</code> object with every header definition properties.
+   *
+   * @param type                      The type name for this header definition.
+   * @param firstLine                 The string to output before the content of the first line of this header.
+   * @param beforeEachLine            The string to output before the content of each line of this header (except
+   *                                  firstLine and endLine).
+   * @param endLine                   The string to output before the content of the last line of this header.
+   * @param afterEachLine             The string to output after the content of each line of this header (except
+   *                                  firstLine and endLine).
+   * @param skipLinePattern           The pattern of lines to skip before being allowed to output this header or null
+   *                                  if it can be outputted from the line of the file.
+   * @param firstLineDetectionPattern The pattern to detect the first line of a previous header.
+   * @param otherLineDetectionPattern The pattern to detect any following line of a previous header.
+   * @param lastLineDetectionPattern  The pattern to detect the last line of a previous header.
+   * @throws IllegalArgumentException If the type name is null.
+   */
+  public HeaderDefinition(String type,
+                          String firstLine, String beforeEachLine,
+                          String endLine, String afterEachLine,
+                          String skipLinePattern,
+                          String firstLineDetectionPattern, String otherLineDetectionPattern, String lastLineDetectionPattern,
+                          boolean allowBlankLines, boolean isMultiline, boolean padLines) {
     this(type);
     this.firstLine = firstLine;
     this.beforeEachLine = beforeEachLine;
     this.endLine = endLine;
     this.afterEachLine = afterEachLine;
     this.skipLinePattern = compile(skipLinePattern);
     this.firstLineDetectionPattern = compile(firstLineDetectionPattern);
+    this.otherLineDetectionPattern = compile(otherLineDetectionPattern);
```

Comment:
you would need to update `setPropertyFromString(...)` too for xml header definition.